### ISSUE TITLE
Setting a default value on a filter should provide a UI indication that a filter is selected and being applied

### ISSUE THREAD

**Issue by vesper8** [laravel/nova-issues#3119]
Currently if you add a default value inside a filter using, for example:

```
    public function default()
    {
        return 2;
    }
```

It will select and apply the filter as expected but it will not provide any visual indication that a filter is currently selected and applied, leading to confusion as to why results are being filtered

IMO, if a filter is being applied, it should always cause the filter button to be highlighted and the number next to it to increment based on the number of filters currently active, whether that filter is a result of a user action or a default value

**Comment by crynobone** [comment]
Inspect the implementation code and it clearly mention filter are only applied if it's not the default value. Marking this as `documentation` and should be mentioned in the doc.

**Comment by vesper8** [comment]
@crynobone I realize that this is currently the "intended behaviour" however I opened this issue to voice my concern that this behaviour is not user-friendly, and that if ANY filter is being applied, for any reason, then it should be visibly obvious that a filter is being applied. The fact that this "special" kind of filter, which affects the filter list like a "regular user-selected filter" but isn't visibly represented like a "regular user-selected filter" is, IMO, confusing AF for the end-user. I've already had a client complain and I am unable to find a good middle-ground solution

**Comment by crynobone** [comment]
> IMO, confusing AF for the end-user. I've already had a client complain and I am unable to find a good middle-ground solution

Changing the structure will confuse some other users as well. Making such changes isn't as simple as just adding a new color indication, what should happen with `Reset Filter` with these "Default Filters", this just going to make it confusing even further.

**Comment by vesper8** [comment]
`Reset Filter` should reset all other filters but leave the ones that are hardcoded via default. So if there are two default filters then `Reset Filter` should reset to the filter button being highlighted with a `2` inside of it.

You can still set those filters to 'none' but navigating away and returning, or resetting, will reset to the default filters being applied and visibly indicated

that's how I'd like it to be, that's what I would expect from such a feature, personally

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by davidhemphill** [comment]
@vesper8 I get where you're coming from, but how far does this logic extend? If the developer adds a where constraint on their `indexQuery` method, should that show up in the UI as well since it technically does the same thing as providing a default value to the filter?

**Comment by vesper8** [comment]
@davidhemphill Fair point, but since that kind of filter is done outside of the Filter class then it would be excessive to try to handle that with the filter dropdown. I think it should extend as far as as any filters applied from within the registered filters.. including this particular case being discussed here.

To be honest though I'd like to point out that what brought me here in the first place was that I was asked by a client if I could make filters persist. They say that it requires too many clicks when managing items (add new/enable/disable) from a huge list of items which they first use a filter on. After adding a new item and returning to the list, the filter is no longer active and they have to do multiple clicks to re-enable the filter and continue with their task. I thought I could somehow hack the default method of the filter class and combined with a temporary cache, I could temporarily persist a filter. That's when I ran into the issue of having such filters not be visibly obvious, as opposed to when the client manually sets the filter.. which brought me here.

It would be nice if my suggestion here was implemented, and it would be even nicer if you might consider an easy way of making filters persist, optionally, either for a limited time or for the duration of the current user session.

It doesn't hurt to ask I guess.

Cheers
